packages/typespec-python/test/azure/generated/payload-pageable/payload/pageable/aio/operations/_operations.py

```diff
@@ -180,43 +178,30 @@ def request_query_response_body(
         }
         error_map.update(kwargs.pop("error_map", {}) or {})
 
-        def prepare_request(next_link=None):
-            if not next_link:
-
-                _request = build_server_driven_pagination_continuation_token_request_query_response_body_request(
-                    token=token,
-                    foo=foo,
-                    bar=bar,
-                    headers=_headers,
-                    params=_params,
-                )
-                path_format_arguments = {
-                    "endpoint": self._serialize.url(
-                        "self._config.endpoint", self._config.endpoint, "str", skip_quote=True
-                    ),
-                }
-                _request.url = self._client.format_url(_request.url, **path_format_arguments)
-
-            else:
-                _request = HttpRequest("GET", next_link)
-                path_format_arguments = {
-                    "endpoint": self._serialize.url(
-                        "self._config.endpoint", self._config.endpoint, "str", skip_quote=True
-                    ),
-                }
-                _request.url = self._client.format_url(_request.url, **path_format_arguments)
+        def prepare_request(_continuation_token=None):
 
+            _request = build_server_driven_pagination_continuation_token_request_query_response_body_request(
+                token=_continuation_token,
+                foo=foo,
+                bar=bar,
+                headers=_headers,
+                params=_params,
+            )
+            path_format_arguments = {
+                "endpoint": self._serialize.url("self._config.endpoint", self._config.endpoint, "str", skip_quote=True),
+            }
+            _request.url = self._client.format_url(_request.url, **path_format_arguments)
             return _request
 
         async def extract_data(pipeline_response):
             deserialized = pipeline_response.http_response.json()
-            list_of_elem = _deserialize(List[_models.Pet], deserialized["pets"])
+            list_of_elem = _deserialize(List[_models.Pet], deserialized.get("pets", []))
             if cls:
                 list_of_elem = cls(list_of_elem)  # type: ignore
-            return None, AsyncList(list_of_elem)
+            return deserialized.get("nextToken") or None, AsyncList(list_of_elem)
 
-        async def get_next(next_link=None):
-            _request = prepare_request(next_link)
+        async def get_next(_continuation_token=None):
```

Comment:
same concern. more consistent with `next_link`.

Reply:
I still hope to keep it to avoid potential naming conflict.